samples/README.md

```diff
@@ -10,3 +10,13 @@ See [Program.cs](./Hooks/Program.cs) to see how it looks, or run the sample to s
 cd Hooks
 dotnet run
 ```
+
+## NestedTransactions
+The [NestedTransactions](./NestedTransactions) sample show how the `NestedTransactor` can be used to extend `Starcounter.Database` to support _nested transactions_, something not natively supported.
```

Comment:
We still don't support nested transactions since there is no isolation between a child and a parent. I wonder if we should rename NestedTransactions to something more specific.

Reply:
I actually read some about the context of nested transactions some time ago, and to me it seemed pretty loose what exactly it stipulates. Like if they should be isolated, or not.

But I don't have a strong opinion. If there's a better name, I'm fine adapting.

Reply:
More proper word would be `Reuse` in this case. Even though I agree with @bigwad that `Nested` is confusing, I am not 100% sure that other wording would be better.

Reply:
There is a difference with 2.x nested transactions - in 2.x a failed child fails its parent, which is not perfect but at least prevents changes to bubble up from a failed child. If we make such changes, I'd agree with calling it "nested".

Reply:
I'm not sure I follow that?

```cs
t.Transact(db => {
  var p = db.Insert<Person>();
  t.Transact(_ => throw new Exception());
});
```

Are you saying the above would fail the insert in 2.x but execute it here? If so, I don't see how.

Reply:
In your sample, parent transaction will also fail. Problem start if a parent handles an exception thrown from a child:

```cs
t.Transact(db => {
 try
 {
    t.Transact(db => 
    {  
      var p = db.Insert<Person>();
      throw new Exception());
    }
  }
  catch(Exception){}
});
```
2.x won't insert a Person in this sample. Which is correct as the transaction that inserted it has failed.

Reply:
Aha, thanks for clarifying. Didn't know about that. And agree, that 2.x behavior seem more correct.

Let me consider it some.

Reply:
For this, I vote we open an issue and treat it now as a "known bug" to be aware of, with a note that we'll try improve on this, and abort the transaction just like in 2.x, and we revise it when we have `Rollback` behavior surfaced. WDYT?

Reply:
I'm not sure of the safety of this implementation without proper rollback. Do you think it's useful?

Reply:
Well, useful enough for someone to start testing and using it, "just" being aware of what would be the effect of catching exceptions like that.

But in production code, then maybe not so much I guess. I wouldn't hate postponing this until we have exposed rollback mechanism either.

In practice, I don't think it would differ much, since it's unlikely someone will start using it right away, and reach production without any need of bumping to later version, in where it will be fixed.

Reply:
I for a one hate merging code to develop, which is not production-ready for at least some scenarios. But decide for yourself.

Reply:
Then let's not. We'll hold this until I've had a chance to implement new rollback API.